eng/pipelines/coreclr/superpmi-collect.yml

```diff
@@ -30,8 +30,8 @@ jobs:
     jobTemplate: /eng/pipelines/common/build-coreclr-and-libraries-job.yml
     buildConfig: checked
     platforms:
-    # Linux tests are built on the OSX machines.
-    # - OSX_x64
+    - OSX_x64
```

Comment:
You don't need `OSX_x64` assuming you want to do the collection just for `OSX_arm64`?

Reply:
Well, superpmi-collect-job.yml depends on coreclr__product_build_OSX_x64_checked as you can see [here]( https://github.com/dotnet/runtime/blob/6de7147b9266d7730b0d73ba67632b0c198cb11e/eng/pipelines/coreclr/templates/superpmi-collect-job.yml#L35) 

I think I can make an improvement here and only trigger OSX_x64 `/eng/pipelines/coreclr/templates/build-job.yml` (instead of `build-coreclr-and-libraries-job.yml`) - I don't need OSX_x64 libraries to be built.